crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -381,7 +380,9 @@ impl Struct for DynamicStruct {
 impl Reflect for DynamicStruct {
     #[inline]
     fn type_name(&self) -> &str {
-        &self.name
+        self.represented_type
+            .map(|info| info.type_name())
+            .unwrap_or_default()
```

Comment:
should this be something like:
```suggestion
        self.represented_type
            .map(|info| info.type_name())
            .unwrap_or_else(|| std::any::type_name::<Self>())
```

to aid with debugging and error messages? (likewise for other dynamics).